### ISSUE TITLE
[BUG] Remote SNMP Discovery not working

### ISSUE THREAD

**Issue by alias59** [OCSInventory-NG/OCSInventory-ocsreports#1367]
**OCS Inventory version**
Version : 2.9.2
Distribution : Debian GNU/Linux 11 (bullseye)
PHP Version : 7.4.28
Agent Version: 2.9.1

**Describe the bug**
SNMP Discovery not working

**To Reproduce**
[1284](https://github.com/OCSInventory-NG/OCSInventory-ocsreports/issues/1284)
[357](https://github.com/OCSInventory-NG/UnixAgent/issues/357)

**Expected behavior**
Menu Inventory --> SMMP --> Table should list the servers names discovered by the SNMP Scan
OCS Server should scan (I think) the 192.168.253.207 switch with the configured OID 1.3.6.1.2.1.1.5.0 and should add it to the table like
 
![image](https://user-images.githubusercontent.com/24585721/165747867-a02750bb-4cba-4ed4-86e6-b09493aebbfb.png)

**Screenshots**
![image](https://user-images.githubusercontent.com/24585721/165748624-4acd83a6-f635-4cab-9731-0e0d81df814f.png)

![image](https://user-images.githubusercontent.com/24585721/165750033-a14fc08a-fa96-4903-9845-64a9484f7221.png)

![image](https://user-images.githubusercontent.com/24585721/165748789-b248ac42-0cad-451f-b467-3945f1e436c5.png)

![image](https://user-images.githubusercontent.com/24585721/165748814-b25d6077-996f-492d-b610-6612fca783d8.png)

![image](https://user-images.githubusercontent.com/24585721/165749870-edf817f6-7656-4c8b-8ee5-2c5b54be6970.png)
![image](https://user-images.githubusercontent.com/24585721/165749502-b399787a-c49a-4cd1-93e7-63a09ad89616.png)

I take this opportunity to ask you, what is that?

![image](https://user-images.githubusercontent.com/24585721/165748968-2045f54f-8b9e-42b1-82d6-6768a031876a.png)

Thank you

**Comment by alias59** [comment]
> **OCS Inventory version** Version : 2.9.2 Distribution : Debian GNU/Linux 11 (bullseye) PHP Version : 7.4.28 Agent Version: 2.9.1
> 
> **Describe the bug** SNMP Discovery not working
> 
> **To Reproduce** [1284](https://github.com/OCSInventory-NG/OCSInventory-ocsreports/issues/1284)
> 
> **Expected behavior** Menu Inventory --> SMMP --> Table should list the servers names discovered by the SNMP Scan OCS Server should scan (I think) the 192.168.253.207 switch with the configured OID 1.3.6.1.2.1.1.5.0 and should add it to the table like
> 
> ![image](https://user-images.githubusercontent.com/24585721/165747867-a02750bb-4cba-4ed4-86e6-b09493aebbfb.png)
> 
> **Screenshots** ![image](https://user-images.githubusercontent.com/24585721/165748624-4acd83a6-f635-4cab-9731-0e0d81df814f.png)
> 
> ![image](https://user-images.githubusercontent.com/24585721/165748760-029ee62f-99b7-4645-9dbd-36fddad1ab9b.png)
> 
> ![image](https://user-images.githubusercontent.com/24585721/165748789-b248ac42-0cad-451f-b467-3945f1e436c5.png)
> 
> ![image](https://user-images.githubusercontent.com/24585721/165748814-b25d6077-996f-492d-b610-6612fca783d8.png)
> 
> ![image](https://user-images.githubusercontent.com/24585721/165748134-feadba25-b168-422d-b569-7d87940b8c20.png)
> 
> ![image](https://user-images.githubusercontent.com/24585721/165749502-b399787a-c49a-4cd1-93e7-63a09ad89616.png)
> 
> I take this opportunity to ask you, what is that?
> 
> ![image](https://user-images.githubusercontent.com/24585721/165748968-2045f54f-8b9e-42b1-82d6-6768a031876a.png)
> 
> Thank you

**Comment by alias59** [comment]
Help please ?

**Comment by gillesdubois** [comment]
Hi,

Did you set the network to scan in your SNMP Agent that perform the scan ? In the case the switch is on an other network it might be needed to return value.

Also, in your screen where you run your SNMPWalk you are scanning a device with OID value set to "ProCurve Switch 2810-24G" but your webconsole setup is configured for "ProCruve Switch 2520G..."

 On a first look it seems you didn't configured your type to match the right value.

Regards,
Gilles.

**Comment by alias59** [comment]
Hi @gillesdubois,

Thank for return.
In my configuration OID is the same as given by SNMPWalk.
I got confused with the screenshots.

_Did you set the network to scan in your SNMP Agent that perform the scan ? In the case the switch is on an other network it might be needed to return value._

What do you mean by that ?

**Comment by charleneauger** [comment]
Hi @alias59 ,

Sorry for the delay.

1/ Your `SNMP_NETWORK` configuration value is wrong. You must set a network IP and not the equipment IP. The SNMP feature will scan the newtork IP to find all accessible equipment in this network.

2/ We recommend to set a reconciliation value in your Type configuration. Without this, the SNMP inventory will create a duplicate of the equipment at each scan.

3/ Regarding your last question on Wildcards :  that means that you are not required to set a precise value. For example, you can set `*Switch*` instead of `ProCurve Switch 2810-24G`.

Regards,
Charlene

**Comment by LinekerCalseverini** [comment]
I'm having the same problem.

Using OCS Inventory Version 2.10. The agent configuration:
```
snmptimeout=3
snmp=1
basevardir=/var/lib/ocsinventory-agent
password=<password-hidden>
user=<user-hidden>
debug=1
nosoftware=0
server=http://192.168.99.112:8080/ocsinventory
snmpretry=2
logfile=/var/log/ocsinventory-agent
snmp_switch=on
snmp_network=192.168.99.0
```

In Configuration -> General Configuration:

- IpDiscover ON
- SNMP ON

In Configuration -> SNMP Configuration, I created:

- a Type named "Firewall", with the type condition being the OID 1.3.6.1.2.1.1.1 with the value *FreeBSD*
- a label FireWallDescription
- Associated the OID 1.3.6.1.2.1.1.1 to FireWall Description Label

In the Machine Configuration -> Network Scan, I set the IpDiscover as ON, the SNMP_SWITCH as ON, and the SNMP_NETWORK as 192.168.99.0

In Manage -> Network Scan, I set:

- Subnet 192.168.99.0
- Type Firewall
- SNMP Communities (they are correct, I checked with snmpwalk)

When I run the ocsinventory-agent command, it doesn't discover any devices. Is there something wrong with the configuration?
